dm/dm/master/server.go

```diff
@@ -405,6 +411,45 @@ func subtaskCfgPointersToInstances(stCfgPointers ...*config.SubTaskConfig) []con
 	return stCfgs
 }
 
+func (s *Server) initClusterID(ctx context.Context) error {
+	log.L().Info("init cluster id begin")
+	ctx1, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	resp, err := s.etcdClient.Get(ctx1, common.ClusterIDKey)
+	if err != nil {
+		return err
+	}
+
+	// New cluster, generate a cluster id and backfill it to etcd
+	if len(resp.Kvs) == 0 {
+		ts := uint64(time.Now().Unix())
+		clusterID := (ts << 32) + uint64(rand.Uint32())
+		clusterIDBytes := make([]byte, 8)
+		binary.BigEndian.PutUint64(clusterIDBytes, clusterID)
+		_, err = s.etcdClient.Put(ctx1, common.ClusterIDKey, string(clusterIDBytes))
+		if err != nil {
+			return err
+		}
+		atomic.StoreUint64(&s.clusterID, clusterID)
+		log.L().Info("generate and init cluster id success")
+		return nil
+	}
+
+	if len(resp.Kvs[0].Value) != 8 {
+		return terror.ErrMasterInvalidClusterID.Generate(resp.Kvs[0].Value)
+	}
+
+	s.clusterID = binary.BigEndian.Uint64(resp.Kvs[0].Value)
+	log.L().Info("init cluster id success")
+	return nil
+}
+
+// Only leader is able to return correct cluster id
+func (s *Server) ClusterID() uint64 {
+	return atomic.LoadUint64(&s.clusterID)
```

Comment:
please remove `atomic.load/store` and surround it with a rw lock
1.  `atomic.load/store` only states `atomic`. there's no clear definition in atomic package regards to concurrent access, neither `golang memory model`, this load may not load the latest value.
2. not a frequent call which need this kind of optimization.

Reply:
Lock is ok, but i just want an atomic integer, and it had nothing to do with `Premature optimization`, right?

Reply:
seems there's a happens-before relation between atomic operations internally(although not documented), see [this](https://github.com/golang/go/discussions/47141), so it's ok to use it.

recommend to use `go.uber.org/atomic`, it's easy to forget to wrap operation with atomic using `sync/atomic`

Reply:
> seems there's a happens-before relation between atomic operations internally(although not documented), see [this](https://github.com/golang/go/discussions/47141), so it's ok to use it.
> 
> recommend to use `go.uber.org/atomic`, it's easy to forget to wrap operation with atomic using `sync/atomic`

Yes, memory model of go is different from java, and scope of some locks in our code is not clear.